pkg/apis/core/validation/controllerregistration.go

```diff
@@ -44,6 +45,23 @@ func ValidateControllerRegistration(controllerRegistration *core.ControllerRegis
 	return allErrs
 }
 
+// SupportedExtensionKinds contains all supported extension kinds.
+var SupportedExtensionKinds = sets.NewString(
```

Comment:
Can you add a TODO so that we switch/move this back to `extensionsv1alpha1.ExtensionKinds` once https://github.com/gardener/gardener/issues/5270 got resolved?

Reply:
Do you have a concrete use case in mind for relying on this set?
To me it seemed like the `ControllerRegistration` validation right here is the only (sensible) usage.
And then I would argue, that it's actually more correct to put it into the same package like the using validation code.
Otherwise, one might change the set in the extensions API with the (surprising) side-effect that it affects the validation of another API group.

This might be a bit theoretical, but I think, this is the cleaner way to do it :) 
Do you agree?

Reply:
Fair enough for now, I don't have another use case in mind.